File: commonLib/src/main/java/com/travel/commonlib/hotel/CouponModal.java
package com.travel.commonlib.hotel;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

public class CouponModal implements Serializable {

    @SerializedName("UserId")
    @Expose
    private long userId;
    @SerializedName("TransactionType")
    @Expose
    private long transactionType;
    @SerializedName("CouponCodeApplyID")
    @Expose
    private long couponCodeApplyID;
    @SerializedName("TransactionId")
    @Expose
    private long transactionId;
    @SerializedName("Code")
    @Expose
    private String code;
    @SerializedName("Flag")
    @Expose
    private Object flag;
    @SerializedName("Status")
    @Expose
    private String status;
    @SerializedName("discount")
    @Expose
    private double discount;
    @SerializedName("Ispercentage")
    @Expose
    private boolean ispercentage;
    @SerializedName("isCouponCode")
    @Expose
    private boolean isCouponCode;
    @SerializedName("successMessageEn")
    @Expose
    private Object successMessageEn;
    @SerializedName("successMessageAr")
    @Expose
    private Object successMessageAr;
    @SerializedName("isBinEnable")
    @Expose
    private boolean isBinEnable;
    @SerializedName("selectedBinSeries")
    @Expose
    private Object selectedBinSeries;

    public CouponModal() {

    }

    public long getUserId() {
        return userId;
    }

    public void setUserId(long userId) {
        this.userId = userId;
    }

    public long getTransactionType() {
        return transactionType;
    }

    public void setTransactionType(long transactionType) {
        this.transactionType = transactionType;
    }

    public long getCouponCodeApplyID() {
        return couponCodeApplyID;
    }

    public void setCouponCodeApplyID(long couponCodeApplyID) {
        this.couponCodeApplyID = couponCodeApplyID;
    }

    public long getTransactionId() {
        return transactionId;
    }

    public void setTransactionId(long transactionId) {
        this.transactionId = transactionId;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public Object getFlag() {
        return flag;
    }

    public void setFlag(Object flag) {
        this.flag = flag;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public double getDiscount() {
        return discount;
    }

    public void setDiscount(double discount) {
        this.discount = discount;
    }

    public boolean isIspercentage() {
        return ispercentage;
    }

    public void setIspercentage(boolean ispercentage) {
        this.ispercentage = ispercentage;
    }

    public boolean isCouponCode() {
        return isCouponCode;
    }

    public void setCouponCode(boolean couponCode) {
        isCouponCode = couponCode;
    }

    public Object getSuccessMessageEn() {
        return successMessageEn;
    }

    public void setSuccessMessageEn(Object successMessageEn) {
        this.successMessageEn = successMessageEn;
    }

    public Object getSuccessMessageAr() {
        return successMessageAr;
    }

    public void setSuccessMessageAr(Object successMessageAr) {
        this.successMessageAr = successMessageAr;
    }

    public boolean isBinEnable() {
        return isBinEnable;
    }

    public void setBinEnable(boolean binEnable) {
        isBinEnable = binEnable;
    }

    public Object getSelectedBinSeries() {
        return selectedBinSeries;
    }

    public void setSelectedBinSeries(Object selectedBinSeries) {
        this.selectedBinSeries = selectedBinSeries;
    }
}

File: commonLib/src/main/java/com/travel/commonlib/flight/PaymentSubmitModel/CouponModal.kt
package com.travel.commonlib.flight.PaymentSubmitModel

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import com.travel.jumbo.utilities.CommonFunctions

import java.io.Serializable

/**
 * Created by Rohith (Android) on 10/30/2020.
 */
class CouponModal : Serializable {
    @SerializedName("UserId")
    @Expose
    var userId: Int? = null

    @SerializedName("TransactionType")
    @Expose
    var transactionType = 0

    @SerializedName("CouponCodeApplyID")
    @Expose
    var couponCodeApplyID: Int? = null

    @SerializedName("TransactionId")
    @Expose
    var transactionId: Int? = null

    @SerializedName("Code")
    @Expose
    var code: String? = null

    @SerializedName("Flag")
    @Expose
    var flag: String? = null

    @SerializedName("Status")
    @Expose
    var status: Any? = null

    @SerializedName("discount")
    @Expose
    var discount: Double? = null

    @SerializedName("Ispercentage")
    @Expose
    var ispercentage: Boolean? = null

    @SerializedName("isCouponCode")
    @Expose
    private var isCouponCode: Boolean? = null

    @SerializedName("CouponCode")
    @Expose
    private var couponCode: String? = null

    @SerializedName("successMessageEn")
    @Expose
    var successMessageEn: String? = null

    @SerializedName("successMessageAr")
    @Expose
    var successMessageAr: String? = null

    @SerializedName("isBinEnable")
    @Expose
    var binEnable: Boolean? = null

    @SerializedName("selectedBinSeries")
    @Expose
    var selectedBinSeries: Any? = null

    @SerializedName("IsBasefareInclude")
    @Expose
    var basefareInclude: Boolean? = null

    @SerializedName("IsTaxInclude")
    @Expose
    var taxInclude: Boolean? = null

    @SerializedName("CouponLimitAmount")
    @Expose
    var couponLimitAmount: String? = null

    @SerializedName("MobileCode")
    @Expose
    var mobileCode: String? = null

    @SerializedName("PhoneNumber")
    @Expose
    var phoneNumber: String? = null

    @SerializedName("Email")
    @Expose
    var email: Any? = null

    @SerializedName("PaymentGatewayId")
    @Expose
    var paymentGatewayId: String? = null

    @SerializedName("IsLoggedIn")
    @Expose
    var isLoggedIn: String? = null

    @SerializedName("ModalityId")
    @Expose
    var modalityId = 0

    @SerializedName("IsBinSpecific")
    @Expose
    var isBinSpecific = false

    @SerializedName("DiscountAmt")
    @Expose
    var discountAmt = 0.0

    @SerializedName("Currency")
    @Expose
    var currency: String? = null

    @SerializedName("IsApplied")
    @Expose
    var isApplied = false

    @SerializedName("BinSeries")
    @Expose
    var binSeries: Any? = null

    @SerializedName("IsDiscountAppliedCompletely")
    @Expose
    var isDiscountAppliedCompletely = false

    @SerializedName("AlertMessage")
    @Expose
    var alertMessage: String? = null

    @SerializedName("InvalidCouponCode")
    @Expose
    var isInvalidCouponCode = false
    fun setCouponCode(couponCode: Boolean?) {
        isCouponCode = couponCode
    }

    fun getIsBinSpecific(): Boolean {
        return isBinSpecific
    }

    fun setIsBinSpecific(isBinSpecific: Boolean) {
        this.isBinSpecific = isBinSpecific
    }

    fun getCouponCode(): String? {
        return couponCode
    }

    fun setCouponCode(couponCode: String?) {
        this.couponCode = couponCode
    }

    fun IsCouponCode(): Boolean? {
        return isCouponCode
    }

    fun setIsCouponCode(couponCode: Boolean?) {
        isCouponCode = couponCode
    }

    fun getMessage(): String? {
        return when (CommonFunctions.lang) {
            "ar" -> successMessageAr
            "en" -> successMessageEn
            else -> successMessageEn
        }
    }
}